### PR TITLE
Add test to `insert` command for `--version` support

### PR DESCRIPTION
**Pull request by DatCorno**

### CODE REVIEW

**Comment by DatCorno** [comment]
[Add --version for insert](https://trello.com/c/ksQSq6f3/244-add-version-for-insert)

**Comment by DatCorno** [comment]
[Add --version for insert](https://trello.com/c/ksQSq6f3/244-add-version-for-insert)

File: tests/functional/commands/test_insert_command.py [hunk 1]
```diff
@@ -144,3 +144,31 @@ def test_insert_two_hyperparameters(database, monkeypatch):
...
+    orion.core.cli.main(["init_only", "-n", "experiment",
+                         "-c", "./orion_config_random.yaml", script_path, "-x~normal(0,1)"])
+    orion.core.cli.main(["init_only", "-n", "experiment",
+                         "-c", "./orion_config_random.yaml", script_path, "-x~normal(0,1)",
+                         "-y~+normal(0,1)"])
+
+    orion.core.cli.main(["insert", "-n", "experiment", "--version", "1",
+                         "-c", "./orion_config_random.yaml", script_path, "-x=1"])
+
+    exp = list(database.experiments.find({"name": "experiment", "version": 1}))
```
Comment (bouthilx) [inline tests/functional/commands/test_insert_command.py]: From now on we should avoid using the database backend directly in the tests (unless testing the backend itself) and rather use the StorageProtocol interface.

File: tests/functional/commands/test_insert_command.py [hunk 2]
```diff
@@ -144,3 +144,31 @@ def test_insert_two_hyperparameters(database, monkeypatch):
...
+    assert len(exp) == 1
+    exp = exp[0]
+    assert '_id' in exp
+
+    trials = list(database.trials.find({"experiment": exp['_id']}))
```
Comment (bouthilx) [inline tests/functional/commands/test_insert_command.py]: I would simply tests that exp-v1 has 0 trials before insert and 1 after, while exp-v2 has 0 trials before and after insert. No need the check the trials, that's already done in another test.

**Pull request merged**